src/west/manifest.py

```diff
@@ -1309,6 +1331,45 @@ def get_projects(self,
             raise ValueError(unknown, uncloned)
         return ret
 
+    def _get_submodules(self, submodules) -> SubmodulesType:
+        '''Gets a list of `Submodules` objects or boolean from the manifest
+        *submodules*.
+
+        If submodules is empty, a copy of ``self.submodules`` is returned.
+        Otherwise the submodules type is checked and the returned value
+        is bool or list type depending on the check result.
+
+        For submodules of list type, method checks format of all elements
+        and converts dictionaries into the Submodule objects.
+
+        For submodules of bool type, method does nothing, as both boolean
+        values are valid (True - means that all project submodules should be
+        considered and False - means that are not any submodules).
+
+        :param submodules: content of the manifest submodules value.
+        '''
+        self.submodules = submodules
+        self.submodules_type = type(self.submodules)
+        if self.submodules is not None:
```

Comment:
This can be an early return too (which would avoid the double negation as a bonus)